File: hao-yang-finance-app/src/features/Settings/types/company.ts
export interface Company {
	id: string;
	name: string;
	taxId?: string;
	contactPerson?: string;
	phone: string[];
	address?: string;
	email?: string;
	isActive: boolean;
	createdAt: string;
	updatedAt: string;
}

export interface CreateCompanyDto {
	id?: string;
	name: string;
	taxId?: string;
	contactPerson?: string;
	phone: string[];
	address?: string;
	email?: string;
}

export interface UpdateCompanyDto {
	name: string;
	taxId?: string;
	contactPerson?: string;
	phone: string[];
	address?: string;
	email?: string;
	isActive: boolean;
}

File: hao-yang-finance-api/Models/Company.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace hao_yang_finance_api.Models
{
    [Table("company")]
    public class Company
    {
        [Key]
        [Column("id")]
        public string Id { get; set; } = Guid.NewGuid().ToString();

        [Column("name")]
        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;

        [Column("tax_id")]
        [MaxLength(20)]
        public string? TaxId { get; set; }

        [Column("contact_person")]
        [MaxLength(50)]
        public string? ContactPerson { get; set; }


        [Column("address")]
        public string? Address { get; set; }

        [Column("email")]
        [MaxLength(100)]
        public string? Email { get; set; }

        [Column("is_active")]
        public bool IsActive { get; set; } = true;

        [Column("created_at")]
        public string CreatedAt { get; set; } = DateTime.UtcNow.ToString("yyyy-MM-ddTHH:mm:ssZ");

        [Column("updated_at")]
        public string UpdatedAt { get; set; } = DateTime.UtcNow.ToString("yyyy-MM-ddTHH:mm:ssZ");

        // Navigation properties
        public virtual ICollection<Waybill> Waybills { get; set; } = new List<Waybill>();
        public virtual ICollection<Invoice> Invoices { get; set; } = new List<Invoice>();
        public virtual ICollection<CompanyPhone> Phones { get; set; } = new List<CompanyPhone>();
    }
}